src/basic/virt.c

```diff
@@ -485,6 +485,50 @@ int detect_virtualization(void) {
         return r;
 }
 
+int detect_private_users(void) {
+        static thread_local int cached = -1;
+
+        if (cached < 0) {
+                _cleanup_fclose_ FILE *f = NULL;
+                _cleanup_free_ char *buf = NULL;
+                size_t n_allocated = 0;
+                ssize_t n;
+                uint32_t a, b, c;
+                int r;
+
+                f = fopen("/proc/self/uid_map", "re");
+                if (!f)
+                        return log_debug_errno(errno, "Failed to open /proc/self/uid_map: %m");
+
+                n = getline(&buf, &n_allocated, f);
```

Comment:
That'd make it dependent on formatting details in the kernel. To make it robust against alignment changes, it would be necessary to somehow split the line anyway… This is too much bother, I think getline() and sscanf() are more straightforward and robust.